system/jlib/jatomic.hpp

```diff
@@ -23,8 +23,15 @@
 #ifdef _WIN32
 inline static void spinPause() { YieldProcessor(); }
 #else
-#include "xmmintrin.h"
+# include "x86intrin.h"
+# if defined(_ARCH_X86_)
```

Comment:
I don't believe the include should be within the if, though we may need a separate outer if to handle non-intel architectures.